packages/material-ui/src/Tabs/Tabs.test.js

```diff
@@ -133,172 +150,155 @@ describe('<Tabs />', () => {
-    let wrapper;
+    const tabs = (
+      <Tabs width="md" onChange={noop} value={1}>
+        <Tab />
+        <Tab />
+      </Tabs>
+    );
     before(() => {
-      wrapper = shallow(
-        <Tabs width="md" onChange={noop} value={1}>
-          <Tab />
-          <Tab />
-        </Tabs>,
-      );
+      consoleErrorMock.spy();
     });
 
     after(() => {
       consoleErrorMock.reset();
     });
 
     it('should pass selected prop to children', () => {
-      assert.strictEqual(
-        wrapper
-          .find(Tab)
-          .at(0)
-          .props().selected,
-        false,
-        'should have selected to false',
-      );
-      assert.strictEqual(
-        wrapper
-          .find(Tab)
-          .at(1)
-          .props().selected,
-        true,
-        'should have selected',
+      const wrapper = mount(tabs);
+      const tabWrappers = wrapper.find('[role="tab"]').hostNodes();
+
+      assert.isNotOk(tabWrappers.at(0).props()['aria-selected']);
+      assert.isOk(tabWrappers.at(1).props()['aria-selected']);
+    });
+
+    it('should accept any value as selected tab value', () => {
+      const tab0 = {};
+      const tab1 = {};
+      assert.notStrictEqual(tab0, tab1);
+      const wrapper = mount(
+        <Tabs width="md" onChange={noop} value={tab0}>
+          <Tab value={tab0} />
+          <Tab value={tab1} />
+        </Tabs>,
       );
+      const tabWrappers = wrapper.find('[role="tab"]').hostNodes();
+
+      assert.isOk(tabWrappers.at(0).props()['aria-selected']);
+      assert.isNotOk(tabWrappers.at(1).props()['aria-selected']);
     });
 
     it('should switch from the original value', () => {
+      const wrapper = mount(tabs);
       wrapper.setProps({ value: 0 });
-      assert.strictEqual(
-        wrapper
-          .find(Tab)
-          .at(0)
-          .props().selected,
-        true,
-        'should have switched to true',
-      );
-      assert.strictEqual(
-        wrapper
-          .find(Tab)
-          .at(1)
-          .props().selected,
-        false,
-        'should have switched to false',
-      );
+
+      const tabWrappers = wrapper.find('[role="tab"]').hostNodes();
+
+      assert.isOk(tabWrappers.at(0).props()['aria-selected']);
+      assert.isNotOk(tabWrappers.at(1).props()['aria-selected']);
     });
 
     describe('indicator', () => {
       it('should accept a false value', () => {
-        const wrapper2 = mount(
+        const wrapper = mount(
           <Tabs width="md" onChange={noop} value={false}>
             <Tab />
             <Tab />
           </Tabs>,
         );
-        assert.strictEqual(wrapper2.find(TabIndicator).props().style.width, 0);
+        assert.strictEqual(wrapper.find(TabIndicator).props().style.width, 0);
       });
 
-      it('should work server-side', () => {
-        const wrapper2 = shallow(
+      /* wrapping Tabs in forwardRef calls componentDidMount :( 
+      it('should not have style server-side', () => {
```

Comment:
@oliviertassinari What issue is this trying to solve? I see that we currently move the indicator to the `Tab` on the first render but move it into the own render tree after it's mounted. I don't understand what this achieves. Could you elaborate what behavior we're expecting to see? Maybe I can come up with a better test for this behavior then.

Reply:
I believe that we should be using the server-side rendering API of enzyme, shallow was a poor fit for this test case. It's solving 3 problems, the core issue comes from the need to the layout API to position the indicator:
- We want to display the indicator at the right location when doing the server-side rendering. We shouldn't have to wait for the client side reconciliation.
- We don't want to see the indicator to disappear when switching between different pages client-side.
- We want the indicator to be correctly positioned when the font is loaded and changes the tab width.

Reply:
I couldn't find any. What's the ssr rendering API of enzyme?

Reply:
I had https://airbnb.io/enzyme/docs/api/render.html in mind.

Reply:
Will look into it. Doesn't make any statements about lifecycles so not sure this helps.

Reply:
Why mentioning the lifecycles?